src/renovate.ts

```diff
@@ -20,11 +20,15 @@ class Renovate {
   }
 
   async runDockerContainer(): Promise<void> {
+    const group = await this.getDockerGroup();
     const commandArguments = [
       '--rm',
       `--env ${this.configFileEnv}=${this.configFileMountPath()}`,
       `--env ${this.tokenEnv}=${this.token}`,
       `--volume ${this.configFile}:${this.configFileMountPath()}`,
+      `--volume /var/run/docker.sock:/var/run/docker.sock`,
+      `--volume /tmp:/tmp`,
+      `--user 1000:${group}`,
```

Comment:
At second glance at the Docker docs for [USER](https://docs.docker.com/engine/reference/run/#user) it looks like we can just use `user:group` instead of `uid:gid`.

> The followings examples are all valid:
`--user=[ user | user:group | uid | uid:gid | user:gid | uid:group ]`

So we should be able to drop the `getDockerGroup` and just use `1000:docker`, but then whatever 1000 was for obvs.

Reply:
@viceice I did some digging and found out that `1000` is the `runneradmin` user, which you probably already knew. So are you okay with changing this to `--user runneradmin:docker`?

Reply:
We can do this if you are sure, GitHub will never change those numbers.

All renovate images are running also user Ubuntu (1000) with group root (0) by default. The renovate/renovate images also install docker, so they add the docker group (999) as suggested by official docker docu.

GitHub currently uses 116 as docker group id. So if we simply mount the docker socket to the image, the docker group IDs don't macht, so we aren't allowed to use the docket socket from container.

So that's why I override the group for GitHub and make sure that, if the group id changes, it will work as is.

Hopefully I don't forgot something and you can understand my descriptions.

😅

Reply:
No worries, and thanks for the elaborate response :smile: . I've read the Docker docs again and now understand this if for the user inside the container. Lets see if I got this correct:

- user
  - The user ID refers to the user inside the docker container, which you've matched up with the `ubuntu` user (1000) which is used in the renovate images.
  - It is a not relevant that the `runneradmin` user, of the GitHub runner, is 1000 as wel, as this refers to the user inside the docker container.
- group
  - The group inside the docker container must match the GitHub `docker` group (116), that is of the GitHub runner, in order to use the docker socket from inside the container.

Please check if all the above is correct. If so then I suggest we make the argument as follows.

```js
const renovateDockerUser = 'ubuntu';
const githubActionsDockerGroup = 'docker';
```

And then use `--user ${renovateDockerUser}:${githubActionsDockerGroup}` as argument to `docker run`.

Reply:
Yes,sounds right. I'm not a docker expert, just self educated.

I'm not sure if the docker user argument is resolved on host or in container. So we can test it, or if you're sure we can use your approach.

Reply:
> I'm not a docker expert, just self educated.

Haha, same here but I think you're more educated on this topic than I am :wink: 

I've did a quick tests though, by running a docker container and supply a user name that doesn't exist inside the container and I get an error from docker that way. Same with group, but the container still starts with an invalid group but doesn't with an invalid user. Based on this quick test I think we should be able to use the approach I suggested above.
I'll push those changes and then we'll see ;)

Reply:
I'm fairly high hopes the user part will work but less for the docker group as the specific group ID might be required for some reason that I don't know.

Reply:
Ok, I ran the tests and it did actually need the GID of the GitHub runner instead of just a group named `docker`. I've reverted the changes but kept the variable for the Renovate docker user.

I've checked this run and it looks to be working now. I'm happy with it like this so will approve it. Could you please go over my changes to see whether you're happy with it like this and feel free to make changes if you're not. https://github.com/renovatebot/github-action/runs/654130536?check_suite_focus=true